src/controllers/user.js

```diff
@@ -1,7 +1,27 @@
 import User from '../domain/user.js'
 import { sendDataResponse, sendMessageResponse } from '../utils/responses.js'
+import { emailRegex, passwordRegex } from '../utils/regexLib.js'
 
 export const create = async (req, res) => {
+  if (!req.body.email) {
+    return sendDataResponse(res, 400, { email: 'Email is required' })
+  }
+
+  if (!req.body.email.match(emailRegex)) {
+    return sendDataResponse(res, 400, { email: 'Email format invalid' })
+  }
+
+  if (!req.body.password) {
+    return sendDataResponse(res, 400, { email: 'Password is required' })
+  }
+
+  if (!req.body.password.match(passwordRegex)) {
+    return sendDataResponse(res, 400, {
+      email:
+        'Password must contain at least one upper case character, at least one number, at least one special character and not be less than 8 characters in length.'
+    })
+  }
```

Comment:
I think all of this can be moved to a separate validation function that accepts an email and a password, then does one of two things:

- Either it returns nothing or a string message
OR
- It always returns nothing and throws an error with the message on failure, allowing any error handling to process the error